### PR TITLE
Add useful debug info for which website has not been found

### PR DESCRIPTION
**Pull request by cscd98**
### Description (*)
After running Magento setup, with custom modules, you may have created settings for store and websites that do not exist yet.

This creates entries in core_config_data.

When running bin/magento to complete the settings and data migration, Magento will crash with the error message:

```
No such entity.
```

This code adds something useful to the user, informing them it is because the website Id requested does not exist.

### Manual testing scenarios (*)
1. Install Magento 2.3.0
2. Have any module insert data into core_config_data, with website scope values
3. Run bin/magento
4. It will crash with above message

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

**Comment by magento-engcom-team** [comment]
Hi @craigcarnell. Thank you for your contribution 
Here is some useful tips how you can test your changes using Magento test environment.
Add the comment under your pull request to deploy test or vanilla Magento instance:
- `@magento-engcom-team give me test instance` - deploy test instance based on PR changes
- `@magento-engcom-team give me 2.3-develop instance` - deploy vanilla Magento instance

For more details, please, review the [Magento Contributor Assistant](https://devdocs.magento.com/guides/v2.2/contributor-guide/contributing.html#contributor-assist) documentation

**Review by orlangur — CHANGES_REQUESTED** [review]
@craigcarnell I would reword it a bit to something like "Website with ID %d / code %s was not found".

**Comment by cscd98** [comment]
@orlangur please check again

**Review by orlangur — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @orlangur, thank you for the review.
[ENGCOM-3671](https://jira.corp.magento.com/browse/ENGCOM-3671) has been created to process this Pull Request

**Comment by magento-engcom-team** [comment]
Hi @craigcarnell. Thank you for your contribution. 
We will aim to release these changes as part of 2.3.1. 
Please check the release notes for final confirmation.
